src/main/java/org/jvnet/solaris/libzfs/ZFSObject.java

```diff
@@ -503,10 +506,54 @@ public int callback(zfs_handle_t handle, Pointer arg) {
     }
 
     /**
+     * Grants the specified set of permissions to this dataset.
+     * NOTE that the "zfs_perm_*" libzfs.so routines are not present at
+     * all in OpenZFS, nor even in Sun Solaris 10u8 (last seen in 10u6).
+     * Not sure about Sol10u7 since it was a rather intermediate release.
+     * They may have been replaced with other logic, since zfs_allow_t and
+     * related data types and comments are still present in the headers.
+     */
+    public void allow(ACLBuilder acl) {
+        String abi = library.getFeature("LIBZFS4J_ABI_zfs_perm_set");
+        if (abi.equals("pre-sol10u8")) {
+            for (PermissionBuilder b : acl.builders) {
+                if(LIBZFS.zfs_perm_set(handle,b.toNativeFormat(this))!=0)
+                    throw new ZFSException(library);
+            }
+        } else {
+            if (abi == null)
+                abi = "<null>";
+            LOGGER.log(Level.FINE, "libzfs4j::allow() was called while LIBZFS4J_ABI_zfs_perm_set=='"+abi+"'");
+        }
+
+    }
+
+    /**
+     * Revokes the specified set of permissions to this dataset.
+     * See also the detailed "NOTE" comment above.
+     */
+    public void unallow(ACLBuilder acl) {
```

Comment:
Sorry, missed the comment. It will definitely show warnings, though there many warnings to which we do not pay much attention. One of allow/unallow methods was definitely causing Jenkins core build failure in https://github.com/jenkinsci/jenkins/pull/2776

If you feel that the current approach is preferable, I am totally fine with that since there is a proper Disclaimer-alike comment